lib/teleterm/clusters/cluster_auth.go

```diff
@@ -126,13 +127,14 @@ func (c *Cluster) LocalLogin(ctx context.Context, user, password, otpToken strin
 
 // SSOLogin logs in a user to the Teleport cluster using supported SSO provider
 func (c *Cluster) SSOLogin(ctx context.Context, providerType, providerName string) error {
-	if _, err := c.updateClientFromPingResponse(ctx); err != nil {
+	pingResp, err := c.updateClientFromPingResponse(ctx)
```

Comment:
Unless I miss something about the key TTL, I think instead of passing the ping response to `c.login`, you might as well set `c.clusterClient.KeyTTL` directly in `c.updateClientFromPingResponse`.

It was originally made for use cases like this, see 5ca6890.